src/components/GraphicsMenu.tsx

```diff
@@ -180,14 +221,30 @@ export default class GraphicsMenu extends React.Component<
     if (this.state.invalidName) {
       validInputColor = "danger";
     }
+
+    let selectedVariableName = "";
```

Comment:
This block is a bit hackish, ideally there would be a "select variable" event I could catch and update the axis info. Right now if you change the selected variable it doesnt update the axis you can choose from until something causes the menu to re-render.

Reply:
I think the VariableTracker can provide what you need? It has a 'variablesChanged' and a 'selectedVariablesChanged' signal one of which you could catch to perform the update. You'd need to connect the signal in your component constructor or componentDidMount event so it's ready when the component is active: varTracker.selectedVariablesChanged.connect(yourMethodCall)
Then disconnect it when the component is no longer using it. Your method call (which you pass to the signal) will be a handler that is called when the selected variables change. The VarMenu.tsx has an example which is how it updates the variable selections in the UI.

Reply:
I think the VariableTracker can provide what you need? It has a 'variablesChanged' and a 'selectedVariablesChanged' signal one of which you could catch to perform the update. You'd need to connect the signal in your component constructor or componentDidMount event so it's ready when the component is active: varTracker.selectedVariablesChanged.connect(yourMethodCall)
Then disconnect it when the component is no longer using it. Your method call (which you pass to the signal) will be a handler that is called when the selected variables change. The VarMenu.tsx has an example which is how it updates the variable selections in the UI.

Reply:
Ya this is what I was looking for. I'll update it.